File: semester 4/MPP/JFTS/FestivalNetworking/src/main/java/ro/mpp/net/server/FestivalJsonCS.java
package ro.mpp.net.server;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp.authenticator.IAuthenticator;
import ro.mpp.service.IFestivalService;

import java.net.Socket;

public class FestivalJsonCS extends AbstractConcurrentServer {
    private static final Logger logger = LogManager.getLogger(FestivalJsonCS.class);
    private final IFestivalService festivalService;
    private final IAuthenticator authenticator;

    public FestivalJsonCS(int port, IFestivalService festivalService, IAuthenticator authenticator) {
        logger.info("Creating FestivalJsonCS");
        super(port);
        logger.info("Created successfully FestivalJsonCS");
        this.festivalService = festivalService;
        this.authenticator = authenticator;
    }

    @Override
    protected Thread createWorker(Socket client) {
        return new Thread(new FestivalClientJsonWorker(festivalService, authenticator, client));
    }
}

File: semester 4/MPP/C#FTS/FTS.Networking/Networking/Server/FestivalJsonCS.cs
using C_FTS.Networking.Server;

using FTS.Networking.Networking.Service;

using log4net;
using System.Net.Sockets;

namespace FTS.Networking.Networking.Server
{
    public class FestivalJsonCS : ConcurrentServer
    {
        private readonly IFestivalSubject _festivalService;
        private static readonly ILog logger = LogManager.GetLogger(typeof(FestivalJsonCS));

        public FestivalJsonCS(string host, int port, IFestivalSubject festivalService) : base(host, port)
        {
            logger.Info("Creating FestivalJsonCS");
            this._festivalService = festivalService;
            logger.Info("Created FestivalJsonCS successfully");
        }

        protected override void HandleClient(TcpClient client)
        {
            logger.InfoFormat("Handling client: {0}", client.Client.RemoteEndPoint);
            FestivalClientJsonWorker worker = new FestivalClientJsonWorker(this._festivalService, client);
            worker.Run();
        }
    }
}